main.cpp

```diff
@@ -95,6 +96,16 @@ int main(int argc, char** argv)
             untrusted_dir = get_dir(i++);
         else if (a == "--experimental")
             ;
+        else if (a.rfind("-D", 0) == 0)
+        {
+            std::string define = a.substr(2);
+            if (define.empty())
+            {
+                fprintf(stderr, "macro name missing after '-D'\n");
```

Comment:
To be consistent with other cmd-line parameter errors, it is better to print the usage before exiting.